src/libbson/src/bson/bson-json.c

```diff
@@ -622,9 +622,10 @@ _bson_json_read_integer (bson_json_reader_t *reader, uint64_t val, int64_t sign)
 
    if (rs == BSON_JSON_REGULAR) {
       BASIC_CB_BAIL_IF_NOT_NORMAL ("integer");
+      BSON_ASSERT (mlib_in_range (int, len));
```

Comment:
Rather than assert, suggest checking if `len` is greater than `INT_MAX` and returning an error. I expect this applies to all types being read. That may avoid a possible crash if the input JSON has a very large (over INT_MAX length) key.

Here is a commit with a test (which is only run when the environment variable `MONGOC_TEST_LARGE_ALLOCATIONS=ON` is defined) and [suggested change](https://github.com/mongodb/mongo-c-driver/commit/d433c227e44226701b7a823fa4bd30a09ac61bad).